### DISCUSSION TITLE
[Show and Tell] Immortalize your text and files anonymously on-chain

### DISCUSSION THREAD

**Discussion by JackHamer09** [zkSync-Community-Hub/zksync-developers#187] [Show and Tell]
### Project Name

EraLink

### Repo Link

https://github.com/JackHamer09/eralink

### Demo Link (Optional)

https://eralink.netlify.app/

### Technologies Used

Paymaster, Solidity, Vue, Nuxt 3

### Describe your project. What does it do? Why did you create it? Any interesting challenges you faced?

EraLink allows you to store texts or files anonymously on-chain with Web2-like experience (no signing required - leveraging paymasters on Era).

**Use cases:**
- Document Verification: Ensure the authenticity of critical documents by storing them on-chain, making it easy to prove they haven’t been altered.
- Proof of Concept or Idea: Timestamp your unique ideas or concepts, providing evidence of originality at a specified time.
- Permanent Content Sharing: Share important links, texts, or files that remain accessible indefinitely, bypassing traditional web hosting limitations.
- Trustless Data Storage: Store data in a transparent manner where anyone can verify its authenticity without relying on intermediaries.
- Decentralized Archiving: Archive critical information ensuring it’s safeguarded against central points of failure or censorship.

**How it works:**
- You enter the text and press create link button
- Random wallet is being created and asks paymaster to publish your submission
- Paymaster pays for transaction and publishes it on-chain

**Resources Used in Development:**
- [zksync-cli](https://github.com/matter-labs/zksync-cli/): Quick development environment setup including contract templates and a zkSync node.
- [zkSync Era Tutorials](https://era.zksync.io/docs/dev/tutorials/): Guides on paymasters.
- [misfits and magic project](https://github.com/MexicanAce/misfits-and-magic/): A reference for contracts and paymaster usage. thx to @MexicanAce

### What specific feedback are you looking for?

Create your links, eternalize your ideas, photos or anything else on-chain and share it online!

### How can the community help or contribute to your project?

Create your links and share it with your friends and community!

**Comment by daisbryce** [comment]
Great project! Leveraging paymasters on Era for anonymous on-chain storage is a clever approach. I'll try creating some links and share feedback with the community. Keep up the good work!

**Comment by kaaron9** [comment]
EraLink is a decentralized application (dApp) that allows users to store texts or files anonymously on-chain with a Web2-like experience, leveraging paymasters on Era. Users can create links to timestamp their ideas, share content, verify documents, and ensure trustless data storage. The project utilizes technologies such as Paymaster, Solidity, Vue, and Nuxt 3. Check out the demo at https://eralink.netlify.app/ and contribute to the repository at https://github.com/JackHamer09/eralink.

**Comment by golfdutoresste** [comment]
🔗 EraLink - Eternalize your data on-chain with Era's paymaster, ensuring trustless and transparent storage. Share links, texts, files, and more securely!

**Comment by hollinsmic** [comment]
EraLink is a project that allows users to store texts or files anonymously on-chain with a Web2-like experience, leveraging paymasters on Era. It has several use cases such as document verification, proof of concept or idea, permanent content sharing, trustless data storage, and decentralized archiving. The project uses Paymaster, Solidity, Vue, and Nuxt 3 technologies. The developer is looking for users to create links and share them with their friends and community.

**Comment by caojeremyv** [comment]
EraLink is a project that allows users to store texts or files anonymously on-chain with a Web2-like experience. It leverages paymasters on Era to enable users to store data in a transparent manner without relying on intermediaries. The project was developed using Solidity, Vue, Nuxt 3, and paymasters. The developer is looking for feedback on the project and encourages users to create links and share them with their friends and community. The community can help by testing the project and providing feedback.
